core/modules/qdisp/Executive.cc

```diff
@@ -367,8 +355,11 @@ bool Executive::_shouldRetry(int refNum) {
     return true;
 }
 
-ExecStatus& Executive::_insertNewStatus(int refNum,
-                                        ResourceUnit const& r) {
+// Returning a reference instead of a shared pointer is an optimization.
+// It requires that the reference does not outlast the pointer in _statuses.
+// At the time of writing, everything in _statuses lasts until this Executive
+// is deleted.
+ExecStatus& Executive::_insertNewStatus(int refNum, ResourceUnit const& r) {
```

Comment:
This is Daniel's optimization and I felt the shift from shared pointer to return a plain old pointer needed a comment. The code, as is, is valid and faster than using shared pointers, so I am inclined to leave it. I think it's also beyond the scope of this ticket to make that change unless the code was incorrect.